File: frontend/src/types/expense.ts
export type ExpenseCategory =
  | 'insurance'
  | 'tax'
  | 'parking'
  | 'accessories'
  | 'repair'
  | 'cleaning'
  | 'fine'
  | 'other';

export const EXPENSE_CATEGORY_LABELS: Record<ExpenseCategory, string> = {
  insurance: 'Insurance',
  tax: 'Tax / Registration',
  parking: 'Parking',
  accessories: 'Accessories',
  repair: 'Repair',
  cleaning: 'Cleaning',
  fine: 'Fine',
  other: 'Other',
};

export interface Expense {
  id: string;
  bike_id: string;
  date: string;
  category: ExpenseCategory;
  cost: number;
  description?: string;
  receipt_image_url?: string;
  notes?: string;
  created_at: string;
}

export interface ExpenseCreate {
  date: string;
  category: ExpenseCategory;
  cost: number;
  description?: string;
  notes?: string;
}

export type ExpenseUpdate = Partial<ExpenseCreate>;

File: backend/app/schemas/expense.py
from uuid import UUID
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.models.enums import ExpenseCategory


class ExpenseCreate(BaseModel):
    logged_at: datetime
    category: ExpenseCategory
    cost: float
    description: Optional[str] = None
    notes: Optional[str] = None


class ExpenseUpdate(BaseModel):
    logged_at: Optional[datetime] = None
    category: Optional[ExpenseCategory] = None
    cost: Optional[float] = None
    description: Optional[str] = None
    notes: Optional[str] = None


class ExpenseResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    bike_id: UUID
    logged_at: datetime
    category: ExpenseCategory
    cost: float
    description: Optional[str]
    receipt_image_url: Optional[str]
    notes: Optional[str]
    created_at: datetime
